src/views/List.jsx

```diff
@@ -1,14 +1,70 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredData, setFilteredData] = useState(data);
+
+	const handleChange = (e) => {
+		const searchTermLocal = e.target.value;
+		//Sets searchTerm to the input value
+		setSearchTerm(searchTermLocal);
+	};
+
+	//useEffect triggered with change in searchTerm or data
+	useEffect(() => {
+		//Debounce function
+		const getItem = setTimeout(() => {
```

Comment:
Optional: The debounce was a nice to have and the PR fulfills all the main requirements so do not feel obligated to spend more time on this.

However the debounce function is not acting on the ideal portion of this component. The input field is firing a `change` event on every keystroke and `handleChange` invokes the state hook directly. As a result the component is reacting to a state change on every keystroke. It is only the filter that kicks in after the debounce times out.

Ideally we would prevent state updates until after the timeout completes

Here is a screenshot illustrating the issue, with one console log placed between lines 6 and 8 and one placed between lines 32 and 34
![Screenshot_20240221_202108](https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/11809481/fb773fdf-07d2-4da2-b340-ab3f972cea69)

Reply:
Again, as Lyle said, the functionality is perfect so this comment is FYI but maybe worth tinkering with so you see what I mean.

We only really _need_ a debounce when querying the back end on an input keystroke but we aren't doing that in this case. The list item data won't change on this page. The filter is happening only on the front end and after the data is loaded. We _do_ want state to update on each keystroke because we want the UI to update as we filter items. 

In fact, we don't need this `useEffect` at all and we don't need to set state for the filtered results. The `searchTerm` is being updated on each keystroke and the `filteredResults` are using the `searchTerm` state.

The hook will only run if the `listpath` changes. In the List view it only updates on the first render. In this video you can see the logs labeled for the hook and for the List. The List updates as expected but the hook never logs again after the first render. This is after removing the useEffect and mapping off of the `filteredResults` const directly.


https://github.com/the-collab-lab/tcl-70-smart-shopping-list/assets/47290951/eb7d0607-2c0f-4f70-95eb-ff82e93c6f6a